### ISSUE TITLE
":not" selector returning too many results

### ISSUE THREAD

**Issue by philipp-classen** [WebReflection/linkedom#126]
In the following example, which uses a `:not` selector, Chrome and Firefox find two matches. However, linkedom returns all three `<p>` elements (effectively ignoring the `:not`):

```
const { parseHTML } = require('linkedom');

const {
  document
} = parseHTML(`
  <html>
    <head></head>
    <body>
      <p>
        <table>
          <tbody>
            <tr>
              <td>
                <p></p>
              </td>
            </tr>
          </tbody>
        </table>
      </p>
      <p></p>
    </body>
  </html>
`);

const result = document.querySelectorAll('p:not(table p)');
console.log(Object.entries(result));
```

Expection: should find two elements
Actual behavior: the result includes all three `<p>` elements

Tested with version 0.14.5.
Node: v16.14.2

---

As a side-note: the example originates from a related [bug in jsdom](https://github.com/jsdom/jsdom/issues/3340), but fails in a different way: while linkedom filters out none of the elements (returning 3 instead of 2), jsdom filters all of them (returning 0 instead of 2). That bug report also includes a JSFiddle to check how the browser matches it: https://jsfiddle.net/7v8dzgfh/6/

**Comment by WebReflection** [comment]
The selector engine is 3rd party … if they don’t support :not there’s little I can do but you could file a big there?

**Comment by philipp-classen** [comment]
> The selector engine is 3rd party … if they don’t support :not there’s little I can do but you could file a bug there?

Reported here: https://github.com/fb55/css-select/issues/742

**Comment by fb55** [comment]
Tl;dr for my response of https://github.com/fb55/css-select/issues/742: This is intended behavior. The root problem is HTML's error correction behavior, which can be confusing.

**Comment by WebReflection** [comment]
@fb55 thank you for double checking this, I indeed didn't see the `table` inside a `p`, where that's not accepted by standard (block contents inside Ps)

@philipp-classen I fully agree with @fb55 here there's an issue in the layout itself, and it's not a goal of this project to fix broken layout, so this is likely a won't ever fix / don't care issue for this project and I hope reasons are good enough for you too.
